obsah/__init__.py

```diff
@@ -242,7 +242,7 @@ def allow_extra_vars():
         """
         Whether to allow --extra-vars parameter to be automatically added
         """
-        return True
+        return os.environ.get('OBSAH_ALLOW_EXTRA_VARS', True)
```

Comment:
data from the environment is represented as strings, and "0" is True-thy in Python (and Ruby).

You'd need to do something like
```python
if value := os.environ.get('OBSAH_ALLOW') is not None:
    return value.lower() in ['true', '1']
return True
```